src/schema.ts

```diff
@@ -14,17 +16,39 @@ export class Schema {
    * @param data - a set of raw data
    * @return a Schema object
    */
-  public static build(data: any): Schema {
+  public static build(data: any, opt: QueryConfig = {}): Schema {
+    opt = extend({}, DEFAULT_QUERY_CONFIG, opt);
+
     // create profiles for each variable
     var summaries: Summary[] = summary(data);
     var types = inferAll(data); // inferAll does stronger type inference than summary
 
     var fieldSchemas: FieldSchema[] = summaries.map(function(summary) {
       var field: string = summary.field;
       var primitiveType: PrimitiveType = types[field] as any;
-      var type: Type = (primitiveType === PrimitiveType.NUMBER || primitiveType === PrimitiveType.INTEGER) ? Type.QUANTITATIVE:
-        primitiveType === PrimitiveType.DATE ? Type.TEMPORAL : Type.NOMINAL;
-
+      var distinct: number = summary.distinct;
+      var type: Type;
+      if (primitiveType === PrimitiveType.NUMBER) {
+        type = Type.QUANTITATIVE;
+      } else if (primitiveType === PrimitiveType.INTEGER) {
+        // use ordinal or nominal when cardinality of integer type is relatively low
+        if (distinct / summary.count < opt.numberOrdinalProportion) {
+          // use nominal if the integers are 1,2,3,...,N where N = cardinality
+          var dataPoints: number[] = keys(summary.unique) as any;
+          var min = dataPoints[0], max = dataPoints[0];
+          for (var i = 1; i < dataPoints.length; i++) {
```

Comment:
I guess you want to support 0, ..., N-1 case too?  

In any case, if you don't want to check each of them one-by-one and wanna use only `min` and `max` -- we already have `summary.min` and `summary.max` so this for-loop seem unnecessary.